### ISSUE TITLE
missing pieces in the AST: parser seems to skip broken annotation value when there is a `.`

### ISSUE THREAD

**Issue by martinlippert** [eclipse-jdt/eclipse.jdt.core#3156]
Similar to #3155, I am working on code completion based on AST nodes and came across this case for a field declaration:

```
@Value(spring.) private String value1;
```

The resulting AST does not contain any information about the annotation value `spring.` anymore. It just gets removed from the AST, where the field declaration looks like this:

```
@Value private String value1;
```

This works for similar cases quite nicely, for example:

```
@Value(spri) private String value1;
```

or 

```
@Value(spring.pr<*>) private String value1;
```

Both  cases result in nice ASTs containing the information about the annotation value in some way.

(@mickaelistria FYI)

**Comment by mickaelistria** [comment]
Seems fine with Javac-based parsing from https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/tree/dom-with-javac :
```
{CompilationUnit: /eclipsecon-demo-jdt-with-javac/src/main/java/snippet/Snippet.java=package snippet;
public class Snippet {
  @Value(spring.$missing$) private String value1;
}
}
```

**Comment by stephan-herrmann** [comment]
keep'em coming, @martinlippert :)

**Comment by martinlippert** [comment]
> keep'em coming, @martinlippert :)

Hehe... :-) Will do...

Thanks again for the super fast turnaround here, very very much appreciated. Extremely awesome!

**Comment by stephan-herrmann** [comment]
Are you counting the beverages you owe me? :)

**Comment by martinlippert** [comment]
> Are you counting the beverages you owe me? :)

Yeah, I think this moves towards a crate already, can't count the individual beverages anymore... ;-)
